chain/chain/src/runtime/mod.rs

```diff
@@ -805,7 +806,8 @@ impl RuntimeAdapter for NightshadeRuntime {
                 }
             }
 
-            let mut signer_access_key = None;
+            let mut account_id = None;
+            let mut payer_and_access_key = None;
```

Comment:
let's be consistent here and use triple `(payer, access_key, account_id)` under one option, you can call it something like `payer_info`. this way you will also avoid `account_id.unwrap()` later